xcm/xcm-simulator/fuzzer/src/fuzz.rs

```diff
@@ -60,10 +71,35 @@ decl_test_network! {
 		parachains = vec![
 			(1, ParaA),
 			(2, ParaB),
+			(3, ParaC),
 		],
 	}
 }
 
+// An XCM message that will be generated by the fuzzer through the Arbitrary trait
+struct XcmMessage {
+	// Source chain
+	source: u32,
+	// Destination chain
+	destination: u32,
+	// XCM message
+	message: Xcm<()>,
+}
+
+impl<'a> Arbitrary<'a> for XcmMessage {
+	fn arbitrary(u: &mut Unstructured<'a>) -> Result<Self, Error> {
+		let source: u32 = u.arbitrary()?;
+		let destination: u32 = u.arbitrary()?;
+		let mut encoded_message: &[u8] = u.arbitrary()?;
+		if let Ok(message) =
+			DecodeLimit::decode_with_depth_limit(MAX_XCM_DECODE_DEPTH, &mut encoded_message)
+		{
+			return Ok(XcmMessage { source, destination, message })
+		}
```

Comment:
This will only fuzz correctly-encoded messages, but is it possible in the network to send badly encoded messages? If so, would we want to fuzz bad messages too?

Reply:
It is possible that a node receives a badly encoded message, but it will not get past the decoding stage (just like in this code). If the `DecodeLimit::decode_with_depth_limit` method contains a reachable panic, infinite loop or overflow, the fuzzer should be able to catch it.

Additionally, as we are executing `XcmMessage`s directly, we cannot try to execute a malformed message, due to Rust's strict type system.